### PR TITLE
DBZ-616 Get default length and scale for types

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-616

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresType.java [hunk 1]
```diff
@@ -88,18 +108,8 @@ public boolean equals(Object obj) {
         if (getClass() != obj.getClass())
             return false;
         PostgresType other = (PostgresType) obj;
-        if (elementType == null) {
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresType.java]: Are we totally sure that the name is unique? I guess it is, although it seems that the OID would be the safest thing to base equality on?

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java [hunk 2]
```diff
@@ -294,6 +312,18 @@ private static TypeRegistry initTypeRegistry(Connection db, Map<String, Integer>
...
+    private static int getSize(final TypeInfo typeInfo, final int oid) {
+        int size = typeInfo.getPrecision(oid, TypeRegistry.NO_TYPE_MODIFIER);
+        if (size == 0) {
+            size = typeInfo.getDisplaySize(oid, TypeRegistry.NO_TYPE_MODIFIER);
+        }
+        return size;
+    }
+
+    private static int getScale(final TypeInfo typeInfo, final int oid) {
+        return typeInfo.getScale(oid, -1);
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java]: s/-1/TypeRegistry.NO_TYPE_MODIFIER/?

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java [hunk 3]
```diff
@@ -294,6 +312,18 @@ private static TypeRegistry initTypeRegistry(Connection db, Map<String, Integer>
         return typeRegistryBuilder.build();
     }
 
+    private static int getSize(final TypeInfo typeInfo, final int oid) {
+        int size = typeInfo.getPrecision(oid, TypeRegistry.NO_TYPE_MODIFIER);
+        if (size == 0) {
+            size = typeInfo.getDisplaySize(oid, TypeRegistry.NO_TYPE_MODIFIER);
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java]: So what's the display size? Is it right to fall back to it?

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresType.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresType.java]: Yes

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java (hunk 3 above)
Comment (jpechane) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java]: This is completely taken from postgres JDBC driver, so it is right fallback
https://github.com/pgjdbc/pgjdbc/blob/48b98971d085a7988b67a31cf5ff9fb52c5534e5/pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java#L1515-L1518

Comment (gunnarmorling) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java]: Ok.

**Comment by gunnarmorling** [comment]
Applied and added one more commit with a few fixes. Thanks, @jpechane!

**Pull request merged**